java.lang.IndexOutOfBoundsException: Index: -1

Should it be checked for 0 events count before subtracting -1?
https://github.com/etcd-io/jetcd/blob/45c9fa7fee85dd431836a6d164d5627e4e5f63a3/jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java#L205

```
Caused by: java.lang.IndexOutOfBoundsException: Index: -1
#011at java.util.Collections$EmptyList.get(Collections.java:4454)
#011at io.etcd.jetcd.api.WatchResponse.getEvents(WatchResponse.java:283)
#011at io.etcd.jetcd.WatchImpl$WatcherImpl.onNext(WatchImpl.java:205)
#011at io.etcd.jetcd.WatchImpl$WatcherImpl.onNext(WatchImpl.java:94)
#011at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:421)
#011at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
message repeated 2 times: [ #011at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)]
#011at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:519)
#011... 5 common frames omitted
```

--- a/jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java
+++ b/jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java
@@ -215,7 +215,7 @@ final class WatchImpl implements Watch {
         //
 
         listener.onError(newCompactedException(response.getCompactRevision()));
-      } else if (response.getEventsCount() > 0) {
+      } else if (response.getEventsCount() == 0 && option.isProgressNotify()) {
         
         //
         // Event
@@ -233,6 +233,8 @@ final class WatchImpl implements Watch {
         //   https://coreos.com/etcd/docs/latest/learning/api.html#watch-streams
         //
 
+        listener.onNext(new io.etcd.jetcd.watch.WatchResponse(response));
+      } else if (response.getEventsCount() > 0) {
         listener.onNext(new io.etcd.jetcd.watch.WatchResponse(response));
         revision = response.getEvents(response.getEventsCount() - 1).getKv().getModRevision() + 1;
       }

